src/main/java/com/laytonsmith/core/events/drivers/PlayerEvents.java

```diff
@@ -2579,4 +2598,975 @@ public boolean modifyEvent(String key, Construct value, BindableEvent event) {
+				Target t = Target.UNKNOWN;
+
+				mapEvent.put("player", new CString(pae.getPlayer().getName(), t));
+				mapEvent.put("type", new CString(pae.getAnimationType().name(), t));
+
+				return mapEvent;
+
+			} else {
+				throw new EventException("Cannot convert event to PlayerAnimationEvent.");
+			}
+		}
+
+		@Override
+		public Driver driver() {
+			return Driver.PLAYER_ANIMATION;
+		}
+
+		@Override
+		public boolean modifyEvent(String key, Construct value, BindableEvent event) {
+			return false;
+		}
+	}
+
+	@api
+	public static class player_bucket extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "player_bucket";
+		}
+
+		@Override
+		public String docs() {
+			return "{}"
+					+ "Called when a player interacts with a Bucket."
+					+ "{player : the Player | block : Return the block clicked."
+					+ "| blockface : Get the face on the clicked block."
+					+ "| bucket : Returns the bucket used in this event."
+					+ "| item : Get the resulting item in hand after the bucket event. }"
+					+ "{}"
+					+ "{item}";
+		}
+
+		@Override
+		public Version since() {
+			return CHVersion.V3_3_3;
+		}
+
+		@Override
+		public boolean matches(Map<String, Construct> prefilter, BindableEvent e) throws PrefilterNonMatchException {
+			return true;
+		}
+
+		@Override
+		public BindableEvent convert(CArray manualObject, Target t) {
+			return null;
+		}
+
+		@Override
+		public Map<String, Construct> evaluate(BindableEvent e) throws EventException {
+			if(e instanceof MCPlayerBucketEvent) {
+
+				MCPlayerBucketEvent pbe = (MCPlayerBucketEvent) e;
+				Map<String, Construct> mapEvent = evaluate_helper(e);
+				Target t = Target.UNKNOWN;
+
+				MCBlock block = pbe.getBlockClicked();
+
+				mapEvent.put("player", new CString(pbe.getPlayer().getName(), t));
+				mapEvent.put("block", new CString(block.getType().getName(), t));
+				mapEvent.put("location", ObjectGenerator.GetGenerator().location(block.getLocation(), false));
+
+				CArray face = CArray.GetAssociativeArray(t);
+				face.set("x", new CInt(pbe.getBlockFace().getModX(), t), t);
+				face.set("y", new CInt(pbe.getBlockFace().getModY(), t), t);
+				face.set("z", new CInt(pbe.getBlockFace().getModZ(), t), t);
+				mapEvent.put("blockface", face);
+
+				mapEvent.put("bucket", new CString(pbe.getBucket().getName(), t));
+				mapEvent.put("item", ObjectGenerator.GetGenerator().item(pbe.getItemStack(), t));
+
+				return mapEvent;
+
+			} else {
+				throw new EventException("Cannot convert event to PlayerBucketEvent.");
+			}
+		}
+
+		@Override
+		public Driver driver() {
+			return Driver.PLAYER_BUCKET;
+		}
+
+		@Override
+		public boolean modifyEvent(String key, Construct value, BindableEvent event) {
+			if(event instanceof MCPlayerBucketEvent) {
+				if(key.equalsIgnoreCase("item") && value instanceof CArray) {
+					BukkitMCItemStack is = (BukkitMCItemStack) ObjectGenerator.GetGenerator().item(value, value.getTarget());
```

Comment:
Part 1: Building on what Pseudoknight said, any class that has Bukkit in its name should not be present in function classes.